### PR TITLE
Improve Debug documentation

### PR DESCRIPTION
**Pull request by steveklabnik**

### CODE REVIEW

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -269,6 +269,22 @@ impl<'a> Display for Arguments<'a> {
 
 /// Format trait for the `:?` format. Useful for debugging, all types
 /// should implement this.
+///
+/// Generally speaking, you should just `derive` a `Debug` implementation.
```
Comment (sfackler) [inline src/libcore/fmt/mod.rs]: Probably worth mentioning the debug builder methods for use by manual implementations (`Formatter::debug_<foo>`)

**Comment by alexcrichton** [comment]
I agree with @sfackler that a mention of the debug builders would be good here as well!

File: src/libcore/fmt/mod.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: I have never used these before. Will investigate.

**Comment by steveklabnik** [comment]
updated!

File: src/libcore/fmt/mod.rs [hunk 2]
```diff
@@ -269,6 +269,50 @@ impl<'a> Display for Arguments<'a> {
...
+///     }
+/// }
+///
+/// let origin = Point { x: 0, y: 0 };
+///
+/// println!("The origin is: {:?}", origin);
+/// ```
+///
+/// There are a number of `debug_*` methods on `Formatter` to help you with manual
+/// implemnetations, such as [`debug_struct`][debug_struct].
```
Comment (alexcrichton) [inline src/libcore/fmt/mod.rs]: implemnetations

**Comment by alexcrichton** [comment]
r=me with a minor typo, thanks!

**Comment by steveklabnik** [comment]
Sigh. Confirmed that tests pass now.

**Pull request merged**